src/pages/Reader.tsx

```diff
@@ -30,6 +30,35 @@ class Reader extends React.Component<any, State> {
 
   constructor(props: any) {
     super(props);
+    let readerSettings = {
+      readerMode: ReaderMode.longStrip,
+      imageSizing: ImageSizing.fitHeight,
+      colourTheme: ColourTheme.light,
+      menu: Menu.open,
+    }
+
+    const savedSettings = localStorage.getItem('carrotSettings');
+    if (savedSettings) {
+      try {
+        const parsedSettings = JSON.parse(savedSettings)
+        if (!parsedSettings.readerMode) {

```

Comment:
isnt all these redundant? you already set all the default values of the settings when declaring `readerSettings`

i would rephrase the whole thing as something like:
```
if (parsedSettings?.readerMode !== undefined) {//or whatever null checking u want
  readerSettings.readerMode == parsedSettings.readerMode
}
```
alternatively, you could also store each setting as a seperate localstorage item, although that would probably get too messy

Reply:
no because an evil person would malform the contents of `carrotSettings`, resulting in a failed json parse which is why i try catch it

Reply:
no you can keep the try catch, just the
```
if (!parsedSettings.readerMode)
```
stuff is redundant